### DISCUSSION TITLE
How to build framework for mac osx to run imagemagick in other application

### DISCUSSION THREAD

**Discussion by geenino** [ImageMagick/ImageMagick#4201] [Development]
Hello,

I am trying to use image magick in other application.
It request framework or bundle to execute.

I have seen make file create single mac os executable. which is doesn't work with my application.

can you please help me to build framework.

**Comment by urban-warrior** [comment]
Your questions are not specific enough.  What application? What bundle?  What framework?

**Comment by geenino** [comment]
Sorry for not elaborating the questions.

I am trying you use magick inside Indesign, which i build my own plugin.

They request to use as  bundle or framework format which is for mac os , like dll in windows.

I have found a way to build, which we have to build our own custom bundle to run it.

But when i try to add magick in my xcode project as external library.

My xcode code:
 #include "Magick++/lib/Magick++.h"
#include <png.h>
#include <MagickCore/list.h>

//on main function
Image image;
       
std::string inpath = "/test.png";
std::string outpath = "/testout.jpg";
image.read( inpath );
image.write( outpath);

This throws error as:
**libc++abi.dylib: terminating with uncaught exception of type Magick::ErrorMissingDelegate: Magick: NoDecodeDelegateForThisImageFormat6 `PNG' @ error/constitute.c/ReadImage/574**

If try to use input form jpeg also throw same error as:
NoDecodeDelegateForThisImageFormat6 `JPEG' @ error/constitute.c/ReadImage/574


Can you help me how to solve this problem. how to add delegates to xcode project.

Thank you in advance.
